src/Model/DecisionTreeStep.php

```diff
@@ -168,11 +168,20 @@ public function getAnswersOptionset()
     /**
     * Return the DecisionAnswer rsponsible for displaying this step
     *
-    * @return DecisionTreeAnswer
+    * @return DecisionTreeAnswer|null
     */
     public function getParentAnswer()
     {
-        return DecisionTreeAnswer::get()->filter('ResultingStepID', $this->ID)->First();
+        // Get db ID of the first step in the tree
+        $firstStepId = $this->ParentElement()->FirstStepID;
+
+        // Return the answer only if the selected resulting step id of the doesn't match the first step.
+        // First step can't be used as resulting step
+        if ($firstStepId > 0 && $firstStepId !== $this->ID) {
+            return DecisionTreeAnswer::get()->filter('ResultingStepID', $this->ID)->First();
+        }
+
+        return null;
```

Comment:
It looks like the changes here cause issues with rendering the entire tree when viewing it from a child step